### PR TITLE
Py js read dataframe

### PR DESCRIPTION
**Pull request by myyong**
At this point, the branch enables exporting of python variables `a=1`, but not data frames.

### CODE REVIEW

File: client/src/languagePlugins/python/pythonPlugin.ts [hunk 1]
```diff
@@ -141,6 +141,10 @@ class PythonBlockKind implements Langs.Block {
       let evalCode = "";
       switch(pynode.kind) {
         case 'code': 
+          // this part calls to eval code
+          // get s a json
+          // json contains urls for all data frames
+          // read data from datastore and put into dictionary
           value = "";
```
Comment (tpetricek) [inline client/src/languagePlugins/python/pythonPlugin.ts]: Does this already do something? This looks like it's always setting the value to empty string, without calling the `/eval` end-point of the Python service...

File: client/src/languagePlugins/python/pythonPlugin.ts [hunk 2]
```diff
@@ -167,8 +171,8 @@ class PythonBlockKind implements Langs.Block {
         source: pyBlock.source
 			}
```
Comment (tpetricek) [inline client/src/languagePlugins/python/pythonPlugin.ts:196]: The indentation looks broken here - is it a space vs. tabs issue?

**Comment by myyong** [comment]
The commit https://github.com/wrattler/wrattler/pull/35/commits/a86d776b37529d9fbdd8cd7defecc79609046f97
only works if I am clicking the eval buttons sequentially. Let's say we have 2 cells. You click the 2nd cell, so you have to evaluate antecedents if they're in the first cell. 

I'm having trouble recursively eval-ing the antecedent nodes because they are returning promises. I think we should use something like `await Promise.all(node.antecedents.map(ant => {evaluate(ant)}))` which will chain it all into a single promise. 

But I'm not doing it correctly, so it's not returning promises correctly. Therefore when it comes to evaluating the node.code, it has no access to the node.export.code.value. Any advice for me?

As a note: I've changed the code in the javascript plugin to use async as well, just because in the main eval, we're expecting a promise instead of a value.

**Comment by myyong** [comment]
@tpetricek, thank you! I've resolved and accepted the requests. Commit https://github.com/wrattler/wrattler/commit/e3dd795261c459aba77732f532443e6119ab83fd behaves as we had planned.

**Comment by myyong** [comment]
@tpetricek This commit https://github.com/wrattler/wrattler/commit/b26d21fa87869114dc385897b5924dc724b074e0 addresses issue #38. Could you please check this section particularly. 

It is about accessing a Javascript-created dataframe from Python. Is it correct to pass the values of the antecedent nodes this way?

```
case 'code': 
          var argDictionary:{[key: string]: string} = {}
          let importedVars = "";
          for (var i = 0; i < pyNode.antecedents.length; i++) {
            let imported = <Graph.PyExportNode>pyNode.antecedents[i]
            console.log(imported);
            let value = imported.value.data
            console.log(value);
            importedVars = importedVars.concat(imported.variableName + " = pd.DataFrame(" + JSON.stringify(value) + ")\n");
          }
          // console.log(argDictionary)
          // console.log(importedVars)
          let src = importedVars.concat(pyNode.source)
          console.log(src)
          let hash = Md5.hashStr(src)
			    let body = {"code": src,"hash": hash,"frames": {}}
          return await getEval(body);
```

**Pull request merged**